### PR TITLE
mobile: Add protocol info in the error details

### PR DESCRIPTION
**Pull request by fredyw**
Risk Level: low
Testing: unit tests
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: mobile

### CODE REVIEW

File: mobile/library/common/http/client.cc [hunk 1]
```diff
@@ -454,6 +454,9 @@ void Client::DirectStreamCallbacks::latchError() {
   std::transform(info.responseFlags().begin(), info.responseFlags().end(), response_flags.begin(),
                  [](StreamInfo::ResponseFlag flag) { return std::to_string(flag.value()); });
   error_msg_details.push_back(absl::StrCat("RESPONSE_FLAGS: ", absl::StrJoin(response_flags, ",")));
+  if (info.protocol().has_value()) {
+    error_msg_details.push_back(absl::StrCat("PROTOCOL: ", *info.protocol()));
```
Comment (abeyad) [inline mobile/library/common/http/client.cc:458]: a comment might be nice pointing to the fact that the values come from https://github.com/envoyproxy/envoy/blob/fbce85914421145b5ae3210c9313eced63e535b0/envoy/http/protocol.h#L13, so if someone is looking at the error details and needs to decipher what the protocol value means, they know which enum to go to

**Review by abeyad — APPROVED** [review]

File: mobile/library/common/http/client.cc (hunk 1 above)
Comment (fredyw) [reply mobile/library/common/http/client.cc:459]: Good idea. Done.

**Review by abeyad — APPROVED** [review]

**Comment by abeyad** [comment]
looks like the format check failed

**Comment by fredyw** [comment]
> looks like the format check failed

Fixed.

**Review by abeyad — APPROVED** [review]

**Pull request merged**